pipert2/core/base/pipe.py

```diff
@@ -17,49 +20,59 @@
     pipe methods such as create flows, register routines, notify events
     and more.
 
     """
 
-    def __init__(self, network: Network = QueueNetwork(),
+    events = class_functions_dictionary()
+
+    def __init__(self, event_board: EventBoard = EventBoard(), network: Network = QueueNetwork(),
                  logger: Logger = get_default_print_logger("Pipe"),
-                 data_transmitter: DataTransmitter = BasicTransmitter(),
-                 auto_pacing_mechanism: bool = False):
+                 data_transmitter: DataTransmitter = BasicTransmitter(), auto_pacing_mechanism: bool = False,
+                 run_rpc_cli: bool = False):
         """
         Args:
+            event_board (EventBoard): The EventBoard of the pipe.
             network: Network object responsible for the routine's communication.
             logger: Logger object for logging the pipe actions.
             data_transmitter: DataTransmitter object to indicate how data flows through the pipe by default.
             auto_pacing_mechanism: True if the user want to use auto pacing mechanism.
-
-        Attributes:
-            network: Network object responsible for the routine's communication.
-            logger: Logger object for logging the pipe actions.
-            data_transmitter: DataTransmitter object to indicate how data flows through the pipe by default.
-            flows (dict[str, Flow]): Dictionary mapping the pipe flows to their name.
-            event_board (EventBoard): EventBoard object responsible for the pipe events.
-            routine_synchroniser (RoutinesSynchroniser): Routine synchroniser if auto_pacing_mechanism is true, else None.
-
+            run_rpc_cli: True if the user want to use RPC command line interface.
         """
 
+        super().__init__(event_board, logger)
+        self.event_board = event_board
         self.network = network
         self.logger = logger
-        self.flows = {}
-        self.event_board = EventBoard()
         self.default_data_transmitter = data_transmitter
+        self.run_rpc_cli = run_rpc_cli
+        self.flows = {}
         self.wires: Dict[tuple, Wire] = {}
 
+        if self.run_rpc_cli:
+            self.rpc_server = RPCPipeWrapper(notify_callback=self.event_board.get_event_notifier())
+        else:
+            self.rpc_server = None
+
         if auto_pacing_mechanism:
             self.routine_synchroniser = RoutinesSynchroniser(event_board=self.event_board,
                                                              logger=self.logger,
                                                              notify_callback=self.event_board.get_event_notifier())
         else:
             self.routine_synchroniser = None
 
+    def run_rpc_server(self, endpoint: str):
+        """Binds it to a given endpoint and runs the rpc server.
+
+            Arguments:
+                endpoint: server's endpoint
+        """
+        self.rpc_server.run_rpc_server(endpoint=endpoint)
```

Comment:
I know it's strange, but if a user didn't pass True to use the RPC and call this function, then raise a custom error message or don't call the self.rpc_sever (because it is None instance) in order to avoid pipeline exit (choose what to do yourself, both are fine).